### PR TITLE
DIT-300 - Sample status

### PR DESCRIPTION
**Pull request by aaroncarroll82**
Still WIP currently

### CODE REVIEW

File: app/uk/gov/hmrc/bindingtariffclassification/service/CaseService.scala [hunk 1]
```diff
@@ -34,6 +37,14 @@ class CaseService @Inject()(appConfig: AppConfig,
     caseRepository.insert(c)
   }
 
+  def addInitialSampleStatusIfExists(c: Case): Future[Unit] = {
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/bindingtariffclassification/service/CaseService.scala]: Having discussed the need for the initial "Operator -1" event - only other observation is that maybe this method could be private and called from the 'insert' method above (as that is the only time 'addInitialSampleStatusIfExists' needs to be called - when creating a new case.

File: app/uk/gov/hmrc/bindingtariffclassification/controllers/CaseController.scala [hunk 2]
```diff
@@ -44,6 +44,7 @@ class CaseController @Inject()(appConfig: AppConfig,
       for {
         r <- caseService.nextCaseReference
         c <- caseService.insert(caseRequest.toCase(r))
+        _ <- caseService.addInitialSampleStatusIfExists(c)
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/bindingtariffclassification/controllers/CaseController.scala]: Not sure about calling methods just for their side effects.   Void methods now seem "odd" after doing scala for a few months; but see what the other devs think.

File: app/uk/gov/hmrc/bindingtariffclassification/model/RESTFormatters.scala [hunk 3]
```diff
@@ -17,58 +17,61 @@
...
-import play.json.extra.Jsonx
+import play.json.extra.{InvariantFormat, Jsonx}
 import uk.gov.hmrc.bindingtariffclassification.model.reporting.ReportResult
 import uk.gov.hmrc.play.json.Union
 
 object RESTFormatters {
 
 
   // `Case` formatters
-  implicit val formatCaseStatus = EnumJson.format(CaseStatus)
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/bindingtariffclassification/model/RESTFormatters.scala]: I fixed the missing type warnings to match changes made to a similar PR this morning

File: test/unit/uk/gov/hmrc/bindingtariffclassification/controllers/CaseControllerSpec.scala [hunk 4]
```diff
@@ -92,6 +93,7 @@ class CaseControllerSpec extends UnitSpec with WithFakeApplication with MockitoS
     "return 201 when the case has been created successfully" in {
       when(caseService.nextCaseReference).thenReturn(successful("1"))
       when(caseService.insert(any[Case])).thenReturn(successful(c1))
+      when(caseService.addInitialSampleStatusIfExists(any[Case])).thenReturn(Future.successful(():Unit))
```
Comment (SteveSugden) [inline test/unit/uk/gov/hmrc/bindingtariffclassification/controllers/CaseControllerSpec.scala]: If the controller/service implementation stays the same then maybe some tests that the addInitialSampleStatusIfExists` service method is/isn't called when required.

File: app/uk/gov/hmrc/bindingtariffclassification/controllers/CaseController.scala (hunk 2 above)
Comment (pllaurado) [reply app/uk/gov/hmrc/bindingtariffclassification/controllers/CaseController.scala]: is a good point Steve but in this case it looks ok.(humble opinion). As long as the unit test verifies that we are calling this method just once with the expected params...

File: app/uk/gov/hmrc/bindingtariffclassification/service/CaseService.scala [hunk 5]
```diff
@@ -34,6 +37,14 @@ class CaseService @Inject()(appConfig: AppConfig,
...
+    if (c.sampleStatus.nonEmpty) {
+      val details = SampleStatusChange(None, c.sampleStatus, None)
+      eventService.insert(Event(UUID.randomUUID().toString, details, Operator("-1",Some(c.application.holder.businessName)), c.reference, Instant.now()))
```
Comment (pllaurado) [inline app/uk/gov/hmrc/bindingtariffclassification/service/CaseService.scala]: Could we make -1 a constant? TRADER_OP = -1

**Pull request merged**